soh/src/code/z_horse.c

```diff
@@ -72,7 +72,9 @@ void func_8006D0EC(GlobalContext* globalCtx, Player* player) {
             Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_HORSE, -25.0f, 0.0f, -1600.0f, 0, -0x4000, 0, 1);
         ASSERT(horseActor != NULL);
     } else if ((globalCtx->sceneNum == gSaveContext.horseData.scene) &&
-               (Flags_GetEventChkInf(0x18) != 0 || DREG(1) != 0)) {
+               (((Flags_GetEventChkInf(0x18) != 0) && (!gSaveContext.n64ddFlag ||
```

Comment:
this is mostly an idea: so i see that sometimes several checks need to happen on rando.. would it make sense to extract checks into some centralized checks manager? so then in game code we can just do something like:
```c++
if (normalChecks && rando_canSpawnHorse())?
```

then 1. it's in a centralized place where people know to look for checks but also 2. you can give them a lot more descriptive names than they have now where you basically have to write lots of comments for it to be clear what you're checking and why -- while giving you a place to add those comments :)

will also make syncs with zret a bit easier ;)

Reply:
definitely a good idea! would you think as a generalized pattern it makes sense to have `rando_canXYZ` checks include the rando check in them? so basically the first line of each one of those methods would be `if (!gSaveContext.n64ddFlag) return true;`

it also gets a little complicated based on where those methods would live and what information is available/what would need to be passed is as args, and there are definitely times where we want `normalChecks && randoChecks` and other times where we want `normalChecks || randoChecks`

i'm thinking this would be a good thing to make an issue for so we have time to think through all of the checks we're cleaning up